### DISCUSSION TITLE
Integrate deck.gl with cesium

### DISCUSSION THREAD

**Discussion by federicomarcantognini** [visgl/deck.gl#8415] [Q&A]
Hi all,
any recommendations on how to integrate deck.gl visualization layer in cesium 2/2.5/3D?
I tried a few options mostly to synchronize both cameras but I cannot get these two to play nicely together.

My attempt:
`
    // Initialize a simple deck.gl scatterplot layer
    const scatterplotLayer = new ScatterplotLayer({
      id: 'scatterplot-layer',
      data: [
        { position: [35.02277343658558, 28.134965098706523], size: 10000, color: [255, 0, 0] },
        { position: [-118.2437, 34.0522], size: 500, color: [0, 255, 0] }, // Los Angeles
        { position: [-87.6298, 41.8781], size: 800, color: [0, 0, 255] }, // Chicago
      ],
      getPosition: d => d.position,
      getRadius: d => d.size,
      getColor: d => d.color,
    });

    // Initialize a deck.gl instance with the scatterplot layer
    const deckInstance = new Deck({
      layers: [scatterplotLayer],
    });

    // Handle camera synchronization
    viewer.camera.changed.addEventListener(() => {
      const { positionCartographic, direction } = viewer.camera;

      console.log("viewer.camera", viewer.camera, viewer.camera.positionCartographic, direction)

      deckInstance.setProps({
        viewState: {
          latitude: Math.toDegrees(positionCartographic.latitude),
          longitude: Math.toDegrees(positionCartographic.longitude),
          zoom: positionCartographic.height,
          bearing: Math.toDegrees(viewer.camera.heading),
          pitch: Math.toDegrees(viewer.camera.pitch),
        },
      });
    });

    // Handle resizing
    window.addEventListener('resize', () => {
      viewer.resize();
      deckInstance.setSize([window.innerWidth, window.innerHeight]);
    });`

Any heads up is appreciated.
thank you

**Comment by charlieforward9** [comment]
The primary way to integrate with the Cesium ecosystem is via [Ion hosted tilesets through  `Tile3DLayer`](https://deck.gl/docs/api-reference/geo-layers/tile-3d-layer#load-3d-tiles-from-cesium-ion) - where `Deck` owns the view.

This usage you're expressing here appears to be akin to a 'basemap' integration [react-google-maps](https://vis.gl/frameworks#:~:text=with%20deck.gl.-,react%2Dgoogle%2Dmaps,-A%20React%20wrapper) or [react-map-gl (mapbox)](https://vis.gl/frameworks#:~:text=kepler.gl%20visualizations.-,react%2Dmap%2Dgl,-A%20React%20wrapper)? 

I did a little digging yesterday and am growing on incredibly capable the Cesium Globe is out of the box. I am curious for some thoughts from more seasoned developers on the challenges of creating these integrations. Obviously the bug surface will grow - and maintancen is a lot of work - has this particular integration been discussed before? 

My [recent GlobeView work](https://github.com/visgl/deck.gl/pulls?q=is%3Apr+author%3Acharlieforward9+globe+assignee%3Acharlieforward9+) is specifically aiming to push Deck closer to Cesium feature parity without rebuilding my renderer around a globe-only framework. But, I am also just one guy trying to build smart visualization value in a mature community - big lift for one dude.

🙏🏽 

@donmccurdy @ibgreen
